test/multipart_test.gleam

```diff
@@ -0,0 +1,83 @@
+import bucket/complete_multipart_upload
+import bucket/create_multipart_upload
+import bucket/get_object
+import bucket/upload_part
+import gleam/bit_array
+import gleam/httpc
+import gleam/list
+import gleam/string
+import gleeunit/should
+import helpers
+
+pub fn perform_multipart_upload_test() {
+  // Set up a test bucket
+  helpers.delete_existing_buckets()
+  let bucket = "multipart1"
+  helpers.create_bucket(bucket)
+
+  // Initiate a multipart upload
+  let key = "test/myfile"
+  let assert Ok(res) =
+    create_multipart_upload.request(bucket:, key:)
+    |> create_multipart_upload.build(helpers.creds)
+    |> httpc.send_bits
+  let assert Ok(res) = create_multipart_upload.response(res)
+  should.equal(res.bucket, bucket)
+  should.equal(res.key, key)
+  should.not_equal(res.upload_id, "")
+
+  // Upload some parts (they can be sent in parallel)
+  // NOTE: The minimum Part size for multipart upload is 5MiB (hardcoded in minio)
+  let upload_id = res.upload_id
+  let part_size = 5 * 1024 * 1024
+  let part1 = get_random_bytes(part_size)
+  let part2 = get_random_bytes(part_size)
+  let part3 = <<"Goodbye!":utf8>>
+  let parts =
+    [part1, part2, part3]
+    |> list.index_map(fn(body, i) {
+      let part_number = i + 1
+      let assert Ok(res) =
+        upload_part.request(bucket:, key:, upload_id:, part_number:, body:)
+        |> upload_part.build(helpers.creds)
+        |> httpc.send_bits
+      let assert Ok(res) = upload_part.response(res)
+      should.not_equal(res.etag, "")
+      complete_multipart_upload.Part(part_number:, etag: res.etag)
+    })
+
+  // Complete the multipart upload
+  // NOTE: The Parts must be sorted in ascending order for this operation
```

Comment:
What does this mean?

Reply:
It's too confusing. I'm going to remove this note, and perhaps make `complete_multipart_upload.request()` sort the `parts` argument to prevent the error case I'm alluding to in this note.

The reason for sorting is that when calling `complete_multipart_upload.request()`, the `parts` argument must be ordered by the `part_number`, e.g. `[Part(.., part_number: 1), Part(.., part_number: 2), Part(.., part_number: 3)]`, otherwise the S3 server would respond to such request with an error.

Reply:
done in 71ba290